### PR TITLE
Share to Instagram - Use dominant colors; generate sticker

### PR DESCRIPTION
**Pull request by nicoback2**
### Description

https://user-images.githubusercontent.com/36916764/205125922-e4dc4e93-9a98-4b52-be68-725679b5be8e.MP4
Finish up the share to instagram design. 
What's left:
Loading state
Metrics

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/components/core/Text.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ export type TextProps = RNTextProps & {
     | 'secondary'
     | 'neutral'
     | 'neutralLight4'
+    | 'neutralLight2'
```
Comment (dylanjeffers) [inline packages/mobile/src/components/core/Text.tsx:19]: sweet thanks for doing that. fwiw this colors prop needs a rework in general :D

Comment (dylanjeffers) [inline packages/mobile/src/components/core/Text.tsx:19]: sweet thanks for doing that. fwiw this colors prop needs a rework in general :D

File: packages/mobile/src/components/share-drawer/ShareDrawer.tsx [hunk 2]
```diff
@@ -81,17 +91,38 @@ const useStyles = makeStyles(({ palette }) => ({
...
 }))
 
+const stickerLoadedEventEmitter = new EventEmitter()
+const STICKER_LOADED_EVENT = 'loaded' as const
+
+const DEFAULT_DOMINANT_COLORS = ['000000', '434343']
+
 export const ShareDrawer = () => {
   const styles = useStyles()
+  const viewShotRef = useRef() as React.RefObject<ViewShot>
```
Comment (dylanjeffers) [inline packages/mobile/src/components/share-drawer/ShareDrawer.tsx:109]: all the sticker stuff is dope! is there a way for us to put this logic in a subcomponent, just cause share-drawer is getting pretty unweildy

File: packages/mobile/src/components/share-drawer/ShareDrawer.tsx [hunk 3]
```diff
@@ -134,35 +193,110 @@ export const ShareDrawer = () => {
...
+          reportToSentry({
+            level: ErrorLevel.Error,
+            error: e,
+            name: 'Share to IG Story error - calculate dominant colors step'
+          })
+          toast({ content: messages.shareToStoryError, type: 'error' })
+          return
+        }
+
+        ;[dominantColorHex1, dominantColorHex2] = Array.isArray(
```
Comment (dylanjeffers) [inline packages/mobile/src/components/share-drawer/ShareDrawer.tsx:235]: this line is a lil wild lol,

File: packages/mobile/src/components/share-drawer/ShareDrawer.tsx [hunk 4]
```diff
@@ -134,35 +193,110 @@ export const ShareDrawer = () => {
...
+        return
+      }
+      let dominantColorsResult: Color[]
+      let dominantColorHex1: string
+      let dominantColorHex2: string
+      if (trackImageUri) {
+        try {
+          dominantColorsResult = await getDominantColors(trackImageUri)
+        } catch (e) {
+          reportToSentry({
```
Comment (dylanjeffers) [inline packages/mobile/src/components/share-drawer/ShareDrawer.tsx:226]: thinking this report + toast could maybe be combined into a helper fn?

File: packages/mobile/src/styles/typography.ts [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
 export type FontWeight =
-  | 'heavy'
-  | 'bold'
-  | 'demiBold'
-  | 'medium'
-  | 'regular'
-  | 'light'
-  | 'thin'
-  | 'ultraLight'
+  | 'heavy' // 900
```
Comment (dylanjeffers) [inline packages/mobile/src/styles/typography.ts:4]: so smart hahah thank you

File: packages/mobile/src/utils/convertRGBtoHex.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import type { Color } from '@audius/common'
+
+const componentToHex = (value: number) => {
+  const hex = value.toString(16)
+  return hex.length === 1 ? '0' + hex : hex
+}
+
+export const convertRGBToHex = (color: Color) => {
```
Comment (dylanjeffers) [inline packages/mobile/src/utils/convertRGBtoHex.ts:8]: nicee

File: packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
+import type { Track, User } from '@audius/common'
+import type { StyleProp, ViewStyle } from 'react-native'
+import { Image, View } from 'react-native'
+
+import AudiusLogo from 'app/assets/images/audiusLogoHorizontal.svg'
+import { Divider, Text } from 'app/components/core'
```
Comment (dylanjeffers) [inline packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx:6]: oh cool so we have the share-to-story-sticker subcomponent, yeah i wonder how we might isolate all the share-story logic in share-drawer.... i guess in a hook or smtg?

File: packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+const useStyles = makeStyles(({ palette }) => ({
+  container: {
+    width: 264,
+    backgroundColor: palette.staticWhite,
+    paddingTop: 16,
+    paddingBottom: 18,
+    paddingHorizontal: 12,
+    display: 'flex',
```
Comment (dylanjeffers) [inline packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx:34]: flex is default btw

File: packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+  style?: StyleProp<ViewStyle>
+  /** Called once the image loads successfully */
+  onLoad: () => void
+}
+
+const useStyles = makeStyles(({ palette }) => ({
+  container: {
+    width: 264,
+    backgroundColor: palette.staticWhite,
+    paddingTop: 16,
```
Comment (dylanjeffers) [inline packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx:31]: i think we are trying to use `spacing(4)` in these cases, spacing is an arg in makeStyles

File: packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx (hunk 7 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx:6]: i guess you could have all the share-logic in this component, and render null until user has clicked "share to instagram"?

**Review by dylanjeffers — APPROVED** [review]
this feature is looking great! stoked that the dominant color stuff is working for ya too

File: packages/mobile/src/components/share-drawer/ShareDrawer.tsx (hunk 4 above)
Comment (nicoback2) [reply packages/mobile/src/components/share-drawer/ShareDrawer.tsx:226]: yeah for sure it got really repetitive haha

File: packages/mobile/src/components/share-drawer/ShareDrawer.tsx (hunk 2 above)
Comment (nicoback2) [reply packages/mobile/src/components/share-drawer/ShareDrawer.tsx:109]: yeah i can try that out!

File: packages/mobile/src/components/share-drawer/ShareDrawer.tsx [hunk 10]
```diff
@@ -284,23 +418,40 @@ export const ShareDrawer = () => {
...
+        <ViewShot
+          style={styles.viewShot}
+          ref={viewShotRef}
+          options={{ format: 'png' }}
+        >
+          <ShareToStorySticker
+            onLoad={handleShareToStoryStickerLoad}
+            track={content.track}
+            // TODO(nkang): Not sure if this helps or hurts chances of successful image:
+            // user={account}
```
Comment (sliptype) [inline packages/mobile/src/components/share-drawer/ShareDrawer.tsx:432]: Shouldn't need the `user`, the image hook will use the one in cache. This was only needed for search items where the user wasn't in the cache already

File: packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+type ShareToStoryStickerProps = {
+  track: Pick<Track, 'cover_art_sizes' | 'cover_art' | 'owner_id' | 'title'>
+  user?: Pick<User, 'creator_node_endpoint'>
+  artist: Pick<User, 'user_id' | 'name' | 'is_verified'>
+  style?: StyleProp<ViewStyle>
+  /** Called once the image loads successfully */
+  onLoad: () => void
+}
```
Comment (sliptype) [inline packages/mobile/src/components/share-drawer/ShareToStorySticker.tsx:25]: I'll have something to clean up the image hook types in a coming pr :)

**Review by sliptype — APPROVED** [review]
FIRE

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--generate-images-share-ig

**Pull request merged**